_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,392 @@
+            "headers": {
+                "Authorization": "Bearer ${credential.openAI_key}"
+            },
+            "request_body": "{ \"model\": \"${parameters.model}\", \"prompt\": \"${parameters.prompt}\", \"max_tokens\": ${parameters.max_tokens}, \"temperature\": ${parameters.temperature} }"
+        }
+    ],
+    "access_mode": "public"
+}
+```
+
+If successful, the connector API responds with a `connector_id` and `status` for the connection.
+
+```
+{
+ "connector_id": "M2pBmYgB5cK1H70wk3hd",
+ "status": "CREATED"
+}
+```
+
+## Configuration options
+
+The following configuration options are **required** in order to build a connector:
+
+| Field | Data type | Description |
+| :---  | :--- | :--- |
+| `name` | String | The name of the connector. |
+| `description` | String | The description for the connector. |
+| `version` | Integer | The version for the connector. |
+| `protocol` | String | The protocol for the connection. For AWS services such as SageMaker and Bedrock, use `aws_sigv4`. For all other services, use `http`. |
+| `parameter` | JSON array | The default connector parameters, including `endpoint` and `model`. 
+| `credential` | String | Defines any credential variables required connect to your chosen endpoint. ML Commons uses **AES/GCM/NoPadding** symmetric encryption with a key length of 32 bytes. When a connection cluster first starts, the key persists inside of OpenSearch. Therefore, you do not need to manually encrypt the key.
+| `action` | JSON array | Tells the connector what actions to run after a connection to ML Commons has been made. For more information about how to configure actions, see [Actions](#action-settings).
+
+### Action settings
+
+The `action` setting when creating a connection tells the connector what ML Commons API operation to run against the connection endpoint. You can configure actions using the following settings:
+
+| Field | Data type | Description |
+| :---  | :--- | :--- |
+`action_type` | String | Required. Sets the ML Commons API operation to use upon connection. As of OpenSearch 2.9, only `predict` is supported. 
+`method` | String | Required. Defines the HTTP method for the API call. Supports `POST` and `GET`.
+`url` | String | Required. Sets the connection endpoint the action takes place. This must match the regex expression for the connection using when [adding trusted endpoints](#adding-trusted-endpoints).
+`headers` | Sets the headers used inside the request or response body. Default is `application/json`.
+`request_body` | Required. Sets the parameters contained inside the request body of the action.
+
+## Registering and deploying a connected model
+
+After a connection has been created, use the `connector_id` from the response to register and deploy a connected models.
+
+For registering model, you have the following options:
+
+- You can use `model_group_id` to create the model with an associated model group.
+- If you do not use `model_group_id`, a new model is created.
```

Comment:
Link to the model access control documentation here for more information (_ml-commons-plugin/model-access-control.md).